pkg/features/features.go

```diff
@@ -125,6 +156,14 @@ func (fg featureFlag) SetFeatureFlag(features []string) error {
 	}
 
 	for k, v := range fg {
+		// disable the feature if the requirement does not meet expectations
+		if !checkRequirement(fg, k) {
+			fg[k] = false
+			v = false
+
+			klog.Warningf("Feature gate: %s required: \"%v\" dose not meet expectation, set to disable", k, requirementString(k))
+		}
```

Comment:
Maybe we will never reach that complexity. There are two solution in this case:

1. Calculate the requirement graph in initiation phase
2. The More easier solution: calculate the requirement in `IsEnabled` method

I prefer the first solution. But it will introduce complexity. @z0marlin @akhilerm WDYT ?

Reply:
I also prefer the first solution. Will raise an issue to enhance the feature flag mechanism. I guess for now we will avoid those checks?

Reply:
Refer issue https://github.com/openebs/node-disk-manager/issues/651 for dependent features.